### PR TITLE
feat: Add legal document review use case

### PR DESCRIPTION
**Pull request by leehuwuj**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new contract review use case.
	- Added a README template for the contract review project.
	- Implemented new UI components for uploading contracts and displaying compliance results.
	- Added a new document example for vendor agreements.
	- Introduced a new FastAPI router for modular API organization.

- **Bug Fixes**
	- Enhanced error messages related to agent selection for templates.

- **Documentation**
	- Updated README and comments to reflect new template support and usage instructions.

- **Chores**
	- Added a new `pyproject.toml` for project dependencies and configuration.
	- Streamlined API routing setup in the FastAPI application.

### CODE REVIEW

File: helpers/datasources.ts [hunk 1]
```diff
@@ -26,10 +27,21 @@ export const EXAMPLE_10K_SEC_FILES: TemplateDataSource[] = [
...
   },
 ];
 
+export const EXAMPLE_GDPR: TemplateDataSource = {
+  type: "file",
+  config: {
+    url: new URL(
+      "https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=CELEX:32016R0679",
+    ),
+    filename: "gdpr.pdf",
```
Comment (leehuwuj) [inline helpers/datasources.ts:41]: Add a new field for a better downloaded file name

File: helpers/index.ts [hunk 2]
```diff
@@ -192,7 +193,7 @@ export const installTemplate = async (
     if (
       props.template === "streaming" ||
       props.template === "multiagent" ||
-      props.template === "extractor"
+      props.template === "reflex"
```
Comment (leehuwuj) [inline helpers/index.ts:196]: Changed to `reflex` for sharing reflex code, the specified code for each use case moved to `component/agents` folder; as reflex is only for python so it's a bit tricky, also welcome for better ideas.

File: templates/components/agents/python/contract_review/app/ui/states/workflow.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+
+class GuidelineData(BaseModel):
+    is_completed: bool
+    is_compliant: Optional[bool]
+    clause_text: Optional[str]
+    output: Optional[Dict[str, Any]]
+
+
+class GuidelineHandlerState(rx.State):
```
Comment (leehuwuj) [inline templates/components/agents/python/contract_review/app/ui/states/workflow.py:81]: As the guideline step has a nested data but Reflex is really hard to handle state with complex data so i split it into two models: GuidelineState and GuidelineHandlerState

File: helpers/python.ts [hunk 4]
```diff
@@ -503,6 +485,23 @@ export const installPythonTemplate = async ({
...
+    if (agents) {
+      await copy("**", path.join(root), {
+        parents: true,
+        cwd: path.join(compPath, "agents", "python", agents),
+        rename: assetRelocator,
+      });
+    } else {
+      console.log(
+        red(
+          "There is no agent selected for multi-agent template. Please pick an agent to use via --agents flag.",
```
Comment (marcusschiesser) [inline helpers/python.ts:498]:
```suggestion
          `There is no agent selected for ${template} template. Please pick an agent to use via --agents flag.`,
```

File: questions/questions.ts [hunk 5]
```diff
@@ -354,11 +354,8 @@ export const askProQuestions = async (program: QuestionArgs) => {
     // default to use LlamaParse if using LlamaCloud
     program.useLlamaParse = true;
   } else {
-    // Extractor template doesn't support LlamaParse and LlamaCloud right now (cannot use asyncio loop in Reflex)
-    if (
-      program.useLlamaParse === undefined &&
-      program.template !== "extractor"
-    ) {
+    // Reflex template doesn't support LlamaParse and LlamaCloud right now (cannot use asyncio loop in Reflex)
```
Comment (marcusschiesser) [inline questions/questions.ts:357]: @leehuwuj please add a ticket for this if we continue using more reflex with this PR

File: questions/simple.ts [hunk 6]
```diff
@@ -51,7 +57,7 @@ export const askSimpleQuestions = async (
   let llamaCloudKey = args.llamaCloudKey;
   let useLlamaCloud = false;
 
-  if (appType !== "extractor") {
+  if (appType !== "extractor" && appType !== "contract_review") {
```
Comment (marcusschiesser) [inline questions/simple.ts:60]: why not just?
```suggestion
  if (template !== "reflex") {
```

File: helpers/index.ts (hunk 2 above)
Comment (marcusschiesser) [reply helpers/index.ts:196]: how about just storing those components in `components/reflex`? as we might do something non-agent?

File: questions/questions.ts (hunk 5 above)
Comment (leehuwuj) [reply questions/questions.ts:357]: Sure, I think it's worth logging a ticket for it: https://github.com/run-llama/create-llama/issues/468

File: questions/simple.ts [hunk 7]
```diff
@@ -42,6 +47,7 @@ export const askSimpleQuestions = async (
         },
         { title: "Code Artifact Agent", value: "code_artifact" },
         { title: "Information Extractor", value: "extractor" },
+        { title: "Contract Review", value: "contract_review" },
```
Comment (marcusschiesser) [inline questions/simple.ts:50]:
```suggestion
        { title: "Contract Review (using Workflows)", value: "contract_review" },
```

File: templates/components/reflex/contract_review/app/ui/pages/index.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    route="/",
+    title="Structure extractor",
+)
+def index() -> rx.Component:
+    """The main index page."""
+    return rx.vstack(
+        rx.vstack(
+            rx.heading("Built by LlamaIndex", size="6"),
+            rx.text(
+                "Upload a contract to view the progress of the contract review.",
```
Comment (marcusschiesser) [inline templates/components/reflex/contract_review/app/ui/pages/index.py:22]:
```suggestion
                "Upload a contract to start the review process.",
```

**Review by marcusschiesser — APPROVED** [review]

File: helpers/index.ts (hunk 2 above)
Comment (marcusschiesser) [reply helpers/index.ts:196]: @leehuwuj great idea!

**Pull request merged**